### PR TITLE
Make offset value parameterizable.

### PR DESCRIPTION
**Pull request by BerniVarga**

### CODE REVIEW

File: consumer/consumer.go [hunk 1]
```diff
@@ -42,10 +43,15 @@ type Message struct {
 
 //NewIterator returns a pointer to a freshly created DefaultIterator.
 func NewIterator(config QueueConfig) MessageIterator {
+	offset := "smallest"
+	if len(config.Offset) > 0 {
+		offset = config.Offset;
+	}
```
Comment (martingartonft) [inline consumer/consumer.go]: you would usually see 'if config.Offset != ""' instead of checking the length.

**Comment by martingartonft** [comment]
lgtm

File: consumer/consumer.go (hunk 1 above)
Comment (BerniVarga) [inline consumer/consumer.go]: I've checked for best practices and I found that it's a matter of preference. And both forms are used in Go standard libs. But I can change it.

**Comment by Nesze** [comment]
+1

**Pull request merged**